core/store/src/adapter/chain_store.rs

```diff
@@ -46,6 +46,10 @@ impl StoreAdapter for ChainStoreAdapter {
 
 impl ChainStoreAdapter {
     pub fn new(store: Store, genesis_height: BlockHeight, save_trie_changes: bool) -> Self {
+        let store_genesis_height = get_genesis_height(&store)
+            .expect("Store failed on fetching genesis height")
+            .expect("Genesis height not found in storage");
+        assert_eq!(store_genesis_height, genesis_height, "Genesis height mismatch");
```

Comment:
This is a temp change for verifying correctness.

In part 2 of the PR, we would remove the `genesis_height` param and use store genesis_height directly.

Reply:
Cool, are you planning to space out merging of those two PRs in order to get it some canary exposure? Or are you just relying on tests?

Reply:
Pretty much merging one after another. I don't think we need canary exposure for this as long as initialization works well!